.github/workflows/main.yml

```diff
@@ -13,17 +11,50 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node: [ '10', '12', '14', '16' ]
 
     steps:
       - uses: actions/checkout@v2
 
       - name: Setup node
         uses: actions/setup-node@v2
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
         with:
-          node-version: ${{ matrix.node }}
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Node check
+        run: find . -name "*.js" -type f -not -path "./node_modules/*" -not -path "./\.*" -exec node --check {} \;
+
+      - name: Lint
+        run: yarn && yarn lint
+
+  test:
+    needs: [build]
+    strategy:
+      matrix:
+        os:
+          - "ubuntu-latest"
+        node:
+          - "16"
+          - "14"
+          - "12"
+          - "10"
+          - "8"
```

Comment:
I don't think we need Node 8 here - it wasn't previously in our CI matrix. But if it passes then 🤷  can't hurt.

Reply:
Copying it here:
I added it because we claim to support it:

> Requirements
> Node 8, 10 or higher

We can drop it from the requirements if you prefer this route.

Reply:
Whoops yes, great call. I don't know how I missed that 🤦